### ISSUE TITLE
Change DummyResponse.code to int

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8182]
|<img alt="potato's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| potato reported|
|-|-|
|Trac ID|trac#8182|
|Type|defect|
|Created|2016-01-25 16:21:19Z|
|Branch|https://github.com/twisted/twisted/tree/dummy-code-int-8182|

`twisted.web.test.test_agent.DummyResponse.code` defaults to `b"200"` which is inconsistent as `Response.code` and all uses of it are `int`.

Attachments:

* [DummyResponse-code.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/602/602d180ad067bdabb7291f6dc509c5d962b4e8e6/42f28c4530f8b9f355508d62f7782df87f8d98cf.diff) (448 bytes) - added by potato on 2016-01-25 16:21:42Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__8182 8182
type__defect defect
reporter__potato potato
priority__normal normal
milestone__None None
branch__branches_dummy_code_int_8182 branches/dummy-code-int-8182
branch_author__glyph glyph
status__closed closed
resolution__fixed fixed
component__web web
keywords__None None
time__1453738879144671 1453738879144671
changetime__1453765413764045 1453765413764045
version__None None
owner__glyph glyph
cc__hawkowl
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph commented|
|-|-|

(In [46645]) Branching to dummy-code-int-8182.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph set owner to @glyph|
|-|-|

This is obviously correct, and tests look happy, so I'll land. Thanks for your contribution!

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph set status to `closed`|
|-|-|

(In [46648]) Merge dummy-code-int-8182: Fix attribute of test mock to be the right type.

Author: potato

Reviewer: glyph

Fixes: [#8182](https://github.com/twisted/twisted/issues/8182)

twisted.web.test.test_agent.DummyResponse.code defaults to b"200" which is
inconsistent as Response.code and all uses of it are int.
